atuin-client/src/record/encryption.rs

```diff
@@ -0,0 +1,363 @@
+use atuin_common::record::{AdditonalData, DecryptedData, EncryptedData, Encryption};
+use base64::{engine::general_purpose, Engine};
+use eyre::{ensure, Context, Result};
+use rusty_paserk::{
+    id::EncodeId,
+    wrap::{LocalWrapperExt, Pie},
+};
+use rusty_paseto::core::{
+    ImplicitAssertion, Key, Local, Paseto, PasetoNonce, PasetoSymmetricKey, Payload, V4,
+};
+use serde::{Deserialize, Serialize};
+
+/// Use PASETO V4 Local encryption using the additional data as an implicit assertion.
+#[allow(non_camel_case_types)]
+pub struct PASETO_V4;
+
+/*
+Why do we use a random content-encryption key?
+Originally I was planning on using a derived key for encryption based on additional data.
+This would be a lot more secure than using the master key directly.
+
+However, there's an established norm of using a random key. This scheme might be otherwise known as
+- client-side encryption
+- envelope encryption
+- key wrapping
+
+A HSM (Hardward security module) provider, eg AWS, Azure, GCP, or even a physical device like a yubikey
```

Comment:
Wouldn't wanna leave for the Hacktoberfest 🤣
```suggestion
A HSM (Hardware Security Module) provider, eg: AWS, Azure, GCP, or even a physical device like a YubiKey
```